eta/core/datasets.py

```diff
@@ -683,19 +696,34 @@ def has_data_with_name(self, data_path):
         data_file = os.path.basename(data_path)
         return data_file in self._data_to_labels_map
 
-    def add_file(self, data_path, labels_path=None, new_data_path=None,
-                 new_labels_path=None, move_files=False,
+    def _parse_file_methods(self, file_method):
+        if isinstance(file_method, tuple) and len(file_method) == 2:
+            data_method, labels_method = file_method
+        else:
+            data_method = labels_method = file_method
+
+        if (data_method not in FILE_METHODS
+                or labels_method not in FILE_METHODS):
+            raise ValueError("invalid file_method: %s", str(file_method))
+
+        return _FILE_METHODS_MAP[data_method], _FILE_METHODS_MAP[labels_method]
+
+    def add_file(self, data_path, labels_path=None, new_data_filename=None,
```

Comment:
I get an error when I leave `labels_path` unspecified (see below).  Why are we giving the user the option not to specify `labels_path`?

```python
In [6]: dataset.add_file("/Users/matthew/Data/road_signs/scoop-valid-image-01-tr
   ...: ansformed-labels/data/999.png")                                         
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-e77dc7ffd573> in <module>
----> 1 dataset.add_file("/Users/matthew/Data/road_signs/scoop-valid-image-01-transformed-labels/data/999.png")

~/Code/theta/eta/eta/core/datasets.py in add_file(self, data_path, labels_path, new_data_filename, new_labels_filename, file_method, error_on_duplicates)
    759         new_labels_path = os.path.join(labels_subdir, new_labels_filename)
    760         if labels_path != new_labels_path:
--> 761             labels_method(labels_path, new_labels_path)
    762 
    763         # First remove any other records with the same data filename

~/Code/theta/eta/eta/core/utils.py in copy_file(inpath, outpath, check_ext)
    508         assert_same_extensions(inpath, outpath)
    509     ensure_basedir(outpath)
--> 510     communicate_or_die(["cp", inpath, outpath])
    511 
    512 

~/Code/theta/eta/eta/core/utils.py in communicate_or_die(args, decode)
    320     '''
    321     try:
--> 322         success, out, err = communicate(args, decode=decode)
    323         if not success:
    324             raise ExecutableRuntimeError(" ".join(args), err)

~/Code/theta/eta/eta/core/utils.py in communicate(args, decode)
    293         err: the command's stderr
    294     '''
--> 295     logger.debug("Executing '%s'", " ".join(args))
    296     p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    297     out, err = p.communicate()

TypeError: sequence item 1: expected str instance, NoneType found
```

Reply:
Not sure. I removed it.